io-engine/src/core/env.rs

```diff
@@ -308,43 +315,7 @@ impl MayastorFeatures {
 /// Defaults are redefined here in case of using it during tests
 impl Default for MayastorCliArgs {
     fn default() -> Self {
-        Self {
-            grpc_endpoint: grpc::default_endpoint().to_string(),
-            ps_endpoint: None,
-            ps_timeout: Duration::from_secs(10),
-            ps_retries: 30,
-            node_name: None,
-            env_context: None,
-            reactor_mask: "0x1".into(),
-            mem_size: 0,
-            rpc_address: "/var/tmp/mayastor.sock".to_string(),
-            no_pci: true,
-            log_components: vec![],
-            log_format: None,
-            mayastor_config: None,
-            ptpl_dir: None,
-            pool_config: None,
-            hugedir: None,
-            core_list: None,
-            bdev_io_ctx_pool_size: 65535,
-            nvme_ctl_io_ctx_pool_size: 65535,
-            registration_endpoint: None,
-            nvmf_tgt_interface: None,
-            nvmf_tgt_crdt: [0; TARGET_CRDT_LEN],
-            api_versions: vec![ApiVersion::V0, ApiVersion::V1],
-            diagnose_stack: None,
-            reactor_freeze_detection: false,
-            reactor_freeze_timeout: None,
-            skip_sig_handler: false,
-            enable_io_all_thrd_nexus_channels: false,
-            events_url: None,
-            enable_nexus_channel_debug: false,
-            lvm: false,
-            snap_rebuild: false,
-            developer_delay: false,
-            rdma: false,
-            bs_cluster_unmap: false,
-        }
+        Self::parse_from(std::iter::empty())
```

Comment:
What are we trying to achieve with this change in `Default` ?

Reply:
It seems obvious to me not to copy paste default values but maybe I'm missing the point of this redefinition.

Reply:
Still working on getting everything to build

Reply:
Removed this cause it's not vital to the PR